theseus/optimizer/nonlinear/levenberg_marquardt.py

```diff
@@ -178,13 +175,13 @@ def _check_accept(
         damping_accept: float,
         down_damping_ratio: float,
         up_damping_ratio: float,
+        ellipsoidal_damping: bool,
     ) -> torch.Tensor:
         linearization = self.linear_solver.linearization
-        damping = (
-            self._damping.view(-1, 1)
-            if isinstance(self._damping, torch.Tensor)
-            else self._damping
-        )
+        assert isinstance(self._damping, torch.Tensor)
+        damping = self._damping.view(-1, 1)
+        if ellipsoidal_damping:
+            damping = linearization.diagonal_scaling(damping)
         # Deliberately using Atb before updating the variables, according to
         # the LM reference above
         den = (delta * (damping * delta + linearization.Atb.squeeze(2))).sum(dim=1) / 2
```

Comment:
In trust region method, `den` is also called `predicted reduction`. In addition, a test can be added for `den` as we discussed before.

Reply:
I'll add this to `TrustRegion` class in the Dogleg PR (remind me if I forget). We'll unify these two in a later PR.